### ISSUE TITLE
Azure DB for PostgreSQL collation changes automatically.

### ISSUE THREAD

**Issue by piotrgo** [terraform-providers/terraform-provider-azurerm#396]
Hi, 

We've run into an issue with newly added Azure DB for Postgres. I am not sure whether it's just a misinformation in documentation, or an actual problem with Terraform implementing Azure DB.
DB's collation changes from en_US to en-US and Terraform want's to change it.

### Terraform Version
0.10.5

### Affected Resource(s)
azurerm_postgresql_database

### Terraform Configuration Files
```hcl
resource "azurerm_postgresql_database" "pgsql_db" {
  name                             = "${replace(var.pgsql_srv_name, "-", "_")}_db"
  resource_group_name = "${var.resource_group_name}"
  server_name                 = "${azurerm_postgresql_server.pgsql_srv.name}"
  charset                          = "${var.charset}"
  collation                        = "en_US"
}
```

### Initial Plan Output
```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  + module.pgsql-artifactory.azurerm_postgresql_database.pgsql_db
      id:                             <computed>
      charset:                        "UTF8"
      collation:                      "en_US"
      name:                           "dbname"
      resource_group_name:            "rgname"
      server_name:                    "srvname"

Plan: 1 to add, 0 to change, 0 to destroy.

------------------------------------------------------------------------
```

Apply runs without any problems, immediately after, another plan results with:

```
Terraform will perform the following actions:

-/+ module.pgsql-artifactory.azurerm_postgresql_database.pgsql_db (new resource required)
      id:                             "/subscriptions/ID" => <computed> (forces new resource)
      charset:                        "UTF8" => "UTF8"
      collation:                      "en-US" => "en_US" (forces new resource)
      name:                           "dbname" => "dbname"
      resource_group_name:"rgname" => "rgname"
      server_name:                "srvname" => "srvname"

Plan: 1 to add, 0 to change, 1 to destroy.

------------------------------------------------------------------------


```

### Expected Behavior
DB should be created with specified collation as per PGSQL documentation (underscore in collation type) and no further changes should be planned by Terraform

### Actual Behavior
DB is being created fine, but on subsequent plan it is being marked for a change from en_US to en-US

### Steps to Reproduce
Follow an example from the documentation for setting up Azure DB for PostgreSQL and Azure PostgreSQL DB with collation type set as per PGSQL documentation.

**Comment by tombuildsstuff** [comment]
Hey @piotrgo 

Thanks for opening this issue :)

In the example you've provided above, we're submitting the value `en_US` through to Azure and then getting `en-US` back - out of interest do you see the same issue when using the value `en-US` for `Collation`? Taking a look at our test suite - [we're making use of the collation `English_United States.1252`](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_postgresql_database_test.go#L123) - as such I'd suggest this is a documentation issue

Thanks

**Comment by piotrgo** [comment]
I've run a test, and it behaves as expected when I specify en-US from the start. I did some searching, and Microsoft uses different notation that uses a hyphen instead of underscore.

Do you think it's worth updating the docs in #418?

Cheers

**Comment by tombuildsstuff** [comment]
@piotrgo that's good to hear, thanks for looking into this further - it's definitely worth updating the docs via #418 and I'll close this issue now that's merged. Thanks!

**Comment by stuart-shelton-hmcts** [comment]
When `en-US` is used now, Azure fails the deployment due to an invalid collation.  Using `en_US` works, but it means that Terraform sees a change and deploys the PGSQL infrastructure on every run, destroying any existing data :(

**Comment by katbyte** [comment]
Hey @piotrgo,

Just wanted to let you know we have released [v1.6.0](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/CHANGELOG.md#160-may-24-2018) of the provider hopefully fixing this issue 🙂
